### PR TITLE
fix(VSelect/VMenu): remove duplicate IDs, add aria-activedescendant

### LINKED ISSUE #9680: [Bug Report] v-select accessibility aria-labelledby doesn't work
### Environment
**Vuetify Version:** 2.1.10
**Vue Version:** 2.6.10
**Browsers:** Chrome 78.0.3904.97
**OS:** Windows 10

### Steps to reproduce
use the keyboard to navigate the v-select, when in opened menu use arrow to move from one element to the other

### Expected Behavior
Screen readers should say the current element linked with aria-labeledby

### Actual Behavior
The screen readers says nothing or 'blank' instead of the text
tested on
Chromevox extension: says nothing
NVDA : https://www.nvaccess.org/ says 'blank'

### Reproduction Link
<a href="https://codepen.io/chrmorqmatic/pen/XWWPyoB" target="_blank">https://codepen.io/chrmorqmatic/pen/XWWPyoB</a>

### Other comments
Tested also with official doc on https://vuetifyjs.com/en/customization/accessibility with slots or without, same behavior, same on v-select exemples

### PR DESCRIPTION
**Pull request by KaelWD**
## Description

Removed duplicated IDs in VSelectList caused by inheritAttrs
The v-card styles also make no visual difference so I removed that along with the redundant DOM element
Added aria-activedescendant to menu activators and v-select hidden input, auto ids on v-list-item in menu

## Motivation and Context

fixes #9680

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-menu offset-y>
      <template v-slot:activator="{ attrs, on }">
        <v-btn
          color="primary"
          dark
          v-bind="attrs"
          v-on="on"
        >
          Dropdown
        </v-btn>
      </template>
      <v-list>
        <v-list-item
          v-for="(item, index) in items"
          :key="index"
          :id="`my-custom-id-${index}`"
          @click=""
        >
          <v-list-item-title>{{ item }}</v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>

    <v-select
      :items="items"
      label="Month"
      outlined
    ></v-select>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      items: ['January', 'February', 'March', 'April'],
    })
  }
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

### CODE REVIEW

**Comment by rkrasnan** [comment]
I am doing something wrong I think. 
I didn't get past yarn build.

**Comment by rkrasnan** [comment]
Testing with NVDA. 

**v-menu**
- Menu button is tab accessible. 
- It is expandable with either tab or space. 
- Pressing down reads "January". 
- Pressing down again does not read anything, it should read "February".

**v-select**
- Input is not tab accessible. 
- Clicking the select expands the menu
- Arrow navigation does not work in the expanded menu
- Nothing is read to the screen reader

**Comment by johnleider** [comment]
cc @phiter @KaelWD

**Comment by phiter** [comment]
The issue faced by @rkrasnan not reading the items after the first is due to this playground example generating the same id for every item `my-custom-id`. Using `my-custom-id-${index}` makes it work.

But `v-select` is indeed not working properly on this branch. It's not focusable at all.

**Comment by KaelWD** [comment]
image

File: packages/vuetify/src/components/VAutocomplete/VAutocomplete.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import VTextField from '../VTextField/VTextField'
 
 // Utilities
 import { keyCodes } from '../../util/helpers'
+import mergeData from '../../util/mergeData'
```
Comment (johnleider) [inline packages/vuetify/src/components/VAutocomplete/VAutocomplete.ts]: Should be above line 9

File: packages/vuetify/src/components/VAutocomplete/VAutocomplete.ts [hunk 2]
```diff
@@ -276,12 +277,13 @@ export default VSelect.extend({
...
 
-      input.data = input.data || {}
-      input.data.attrs = input.data.attrs || {}
-      input.data.attrs.autocomplete = input.data.attrs.autocomplete || 'off'
-
-      input.data.domProps = input.data.domProps || {}
-      input.data.domProps.value = this.internalSearch
+      input.data = mergeData(input.data!, {
+        attrs: {
+          'aria-activedescendant': this.$refs.menu && this.$refs.menu.activeTile && this.$refs.menu.activeTile.id,
```
Comment (johnleider) [inline packages/vuetify/src/components/VAutocomplete/VAutocomplete.ts]:
```suggestion
          'aria-activedescendant': getObjectValueByPath(this.$refs.menu, 'activeTile.id', undefined),
```

File: packages/vuetify/src/components/VAutocomplete/VAutocomplete.ts [hunk 3]
```diff
@@ -276,12 +277,13 @@ export default VSelect.extend({
...
+          autocomplete: input.data!.attrs!.autocomplete || 'off',
```
Comment (johnleider) [inline packages/vuetify/src/components/VAutocomplete/VAutocomplete.ts]:
```suggestion
          autocomplete: getObjectValueByPath(input.data!, 'attrs.autocomplete', 'off'),
```

File: packages/vuetify/src/components/VMenu/VMenu.ts [hunk 4]
```diff
@@ -360,6 +373,7 @@ export default baseMixins.extend({
       )
     },
     getTiles () {
+      if (!this.$refs.content) return
```
Comment (johnleider) [inline packages/vuetify/src/components/VMenu/VMenu.ts]: Should have a blank line after

File: packages/vuetify/src/components/VSelect/VSelect.ts [hunk 5]
```diff
@@ -20,6 +20,7 @@ import ClickOutside from '../../directives/click-outside'
 // Utilities
 import { getPropertyFromItem, keyCodes } from '../../util/helpers'
 import { consoleError } from '../../util/console'
+import mergeData from '../../util/mergeData'
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/VSelect.ts]: should be above the other imports.

File: packages/vuetify/src/components/VSelect/VSelect.ts [hunk 6]
```diff
@@ -413,12 +414,18 @@ export default baseMixins.extend<options>().extend({
...
-      input.data!.attrs!.autocomplete = input.data!.attrs!.autocomplete || 'off'
-      input.data!.on!.keypress = this.onKeyPress
+
+      input.data = mergeData(input.data!, {
+        domProps: { value: null },
+        attrs: {
+          readonly: true,
+          type: 'text',
+          'aria-readonly': String(this.readonly),
+          'aria-activedescendant': this.$refs.menu && this.$refs.menu.activeTile && this.$refs.menu.activeTile.id,
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/VSelect.ts]:
```suggestion
          'aria-activedescendant': getObjectValueByPath(this.$refs.menu, 'activeTile.id', undefined),
```

File: packages/vuetify/src/components/VSelect/VSelect.ts [hunk 7]
```diff
@@ -413,12 +414,18 @@ export default baseMixins.extend<options>().extend({
...
+          autocomplete: input.data!.attrs!.autocomplete || 'off',
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/VSelect.ts]:
```suggestion
          autocomplete: getObjectValueByPath(input.data!, 'attrs.autocomplete', 'off'),
```

**Comment by phiter** [comment]
VSelect items are now being read as you navigate through them, but there are some weird issues still.
I talked to Kael on Discord about this, but basically VSelect is reading the input field as "edit blank".

Also if the VSelect is closed and you press arrow to change the item, it just reads "blank" after each keypress.

Here's a video: https://www.youtube.com/watch?v=LdFbpNOrE-k

Here is the proper implementation details with the elements and aria roles to implement an accessible VSelect.

https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html

We could follow some of this.

The implementation uses a button with the currently selected item as a label inside the button, but I don't know if it works well since VSelect extends VInput.

**Comment by KaelWD** [comment]
> activeTile references can be computed
Maybe, but I know refs aren't reactive so I can't be bothered

**Pull request merged**